### ISSUE TITLE
Vi movement keys ( h→ j↓ k↑ l←)

### ISSUE THREAD

**Issue by chkno** [getting-things-gnome/gtg#454]
Vi users stubbornly expect h, j, k, l to move cursors around.  :)

**Comment by chkno** [comment]
I implemented this in a direct but somewhat inelegant way in https://github.com/chkno/gtg/compare/vi-movement

Advice on how to better integrate this is welcome.  (It probably should be configurable and off by default?  Maybe it should be a plugin?)

**Comment by nekohayo** [comment]
Yeah it surely should be a plugin rather than core functionality, not everyone uses the Editor of the Beast :) and personally I can see why it would otherwise be problematic even for myself:
* I expect arrows to work, that's what feels natural to me... hjkl is just as discoverable as ↑↑↓↓←→←→BA  (which _should_ also be a plugin... :)
* I'm a Dvorak typist so hjkl means nothing to me vs, say, dhtn or aoeu or wasd or ,aoe
* I sometimes use typeahead to search the beginning of task titles and use the arrow keys to navigate between results, and would like to be able to use the full alphabet for that instead of hjkl getting hijacked

**Comment by NicTanghe** [comment]
>     * I expect arrows to work, that's what feels natural to me... hjkl is just as discoverable as ↑↑↓↓←→←→BA  (which _should_ also be a plugin... :)

If you need to type and switch lines allot moving your hands over to the arrow keys adds up to alloooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooot of waisted time.

( although on my laptop tats not realy the case)

**Comment by diegogangl** [comment]
This should def be a plugin. However you can't insert things into keypress events through a plugin. This should be done by adding new actions and setting accelerators for them. 
https://stackoverflow.com/questions/46718679/gtk3-how-to-set-cursor-on-a-gtk-treeview-row

You should add the actions to the treeview widget so they are only activated when you press the keys with the treeview in focus. 
Example: https://github.com/getting-things-gnome/gtg/blob/master/GTG/gtk/application.py#L164

You still need to add the accels through the app class though. 

BTW I remember you could add context based shortcuts via CSS. So you can also have bindings like `dd`, `ctG`, etc. I don't know how/if they can be done via Python (CSS bindings were removed in GTK4 so don't use them). Just an idea if you want to take that plugin to the next level :)

**Comment by bepolymathe** [comment]
I think a plugin that allows you to configure moving keyboard shortcuts would be excellent. Especially because there is not only vim as a displacement logic but also bepo, dvorak etc...
